client/service/src/builder.rs

```diff
@@ -771,13 +772,15 @@ where
 		return Err("Warp sync enabled, but no warp sync provider configured.".into())
 	}
 
-	if config.state_pruning.is_archive() {
-		match config.network.sync_mode {
-			SyncMode::Fast { .. } => return Err("Fast sync doesn't work for archive nodes".into()),
-			SyncMode::Warp => return Err("Warp sync doesn't work for archive nodes".into()),
-			SyncMode::Full => {},
-		};
-	}
+	// FIXME
```

Comment:
@andresilva @bkchr

Can there be an appropriate way to access the meta-data of the `StateDb` (or the instance of `StateDb`) from this context?

Reply:
This is not yet exposed anywhere AFAIK. We would probably need to hack around and expose this in the client somewhere. WDYT @andresilva?

Reply:
[Well, that works.](https://github.com/RGafiyatullin/substrate/commit/00c4cbc756b23bb3bda10baefde599c912412ffb)

Although I find it kind of ugly.